### DISCUSSION TITLE
how to import JSON nested with new row for each entry in an JSON object?

### DISCUSSION THREAD

**Discussion by FlaviuRadulescu** [ClickHouse/ClickHouse#39995] [Q&A]
Hi, 

I am new to ClickHouse and I am strugling to import some data from a json document.  I am evaluating ClickHouse for a new project

the document (file) format is like this:

{

    "platform": 12,
    "chart_timestamp": 1653742800,
    "chart_apps": {
        "123": 1,
        "124": 2,
        "125": 3}
},
{

    "platform": 456,
    "chart_timestamp": 1653742123,
    "chart_apps": {
        "156": 2,
        "134": 1,
        }
}

and I am trying to have them inserted in the db format like this:

platform | chart_timestamps | chart_app | chart_position
12         |  1653742800         |  123          | 1
12         |  1653742800         |  124          | 2
12         |  1653742800         |  125          | 3
456       |  1653742123         |  156          | 2
456       |  1653742123         |  134          | 1

I have tried in many ways with but I could not make it work. Can aybody explain to me how I can do it?

 INSERT INTO default.top_charts SELECT JSONExtractString(json, 'platform'), JSONExtractString(json, 'chart_timestamp') FROM file ('/var/lib/clickhouse/user_files/top_charts.json', JSONAsString)

**Comment by UnamedRus** [comment]
https://kb.altinity.com/altinity-kb-queries-and-syntax/jsonextract-to-parse-many-attributes-at-a-time/

+ use arrayJoin (ARRAY JOIN) for parsing of Nested object
